client/components/token-field/index.jsx

```diff
@@ -193,78 +187,22 @@ var TokenField = React.createClass( {
 	},
 
 	_onFocus: function( event ) {
-		if ( this._blurTimeoutID ) {
-			this._clearBlurTimeout();
-			return;
-		} else if ( this.state.isActive ) {
-			return; // already active
-		}
-
 		this.setState( { isActive: true } );
 		if ( 'function' === typeof this.props.onFocus ) {
 			this.props.onFocus( event );
 		}
 	},
 
 	_onBlur: function( event ) {
-		var blurSource = event.relatedTarget ||
-			event.nativeEvent.explicitOriginalTarget || // FF
-			document.activeElement; // IE11
-
-		var stillActive = this.refs.main.contains( blurSource );
-
-		if ( stillActive ) {
-			debug( '_onBlur but component still active; not doing anything' );
-			return; // we didn't leave the component, so don't do anything
-		}
-
-		debug( '_onBlur before timeout setting component inactive' );
-		this.setState( {
-			isActive: false
-		} );
-		/* When the component blurs, we need to add the current text, or
-		 * the selected suggestion (if any).
-		 *
-		 * Two reasons to set a timeout rather than do this immediately:
-		 *  - Some other user action (like tapping on a suggestion) may
-		 *    have caused this blur.  If there is another user-triggered
-		 *    event, we need to give it a chance to complete first.
-		 *  - At one point, using the right arrow key to move the text
-		 *    input was causing a blur to outside the component?! (left
-		 *    arrow key does not do this).  So, we delay the resetting of
-		 *    the state and cancel it if we get focus back quick enough.
-		 */
-		debug( '_onBlur waiting to add current token' );
-		this._blurTimeoutID = window.setTimeout( function() {
-			// Add the current token, UNLESS the text input is empty and
-			// there is a suggested token selected.  In that case, we don't
-			// want to add it, because it's easy to inadvertently hover
-			// over a suggestion.
-			if ( this._inputHasValidValue() ) {
-				debug( '_onBlur after timeout adding current token' );
-				this._addCurrentToken();
-			} else {
-				debug( '_onBlur after timeout not adding current token' );
-			}
-			debug( '_onBlur resetting component state' );
-			this.setState( this.getInitialState() );
-			this._clearBlurTimeout();
-		}.bind( this ), 0 );
-	},
-
-	_clearBlurTimeout: function() {
-		if ( this._blurTimeoutID ) {
-			debug( '_blurTimeoutID cleared' );
-			window.clearTimeout( this._blurTimeoutID );
-			this._blurTimeoutID = null;
-		}
-	},
-
-	_onClick: function() {
-		if ( this.refs.input && ! this.refs.input.hasFocus() ) {
-			debug( '_onClick focusing input' );
-			this.refs.input.focus();
+		debug( '_onBlur setting component inactive' );
```

Comment:
Since the new `_onBlur` code is very simple we can remove this debug statement...